lib/spack/spack/bootstrap.py

```diff
@@ -28,6 +31,208 @@
 import spack.util.path
 from spack.util.environment import EnvironmentModifications
 
+#: Map a bootstrapper type to the corresponding class
+_bootstrap_methods = {}
+
+
+def _bootstrapper(type):
+    """Decorator to register classes implementing bootstrapping
+    methods.
+
+    Args:
+        type (str): string identifying the class
+    """
+    def _register(cls):
+        _bootstrap_methods[type] = cls
+        return cls
+    return _register
+
+
+@_bootstrapper(type='store')
+class _StoreBootstrapper(object):
+    def __init__(self, conf):
+        self.conf = conf
+
+    @staticmethod
+    def try_import(module, abstract_spec):
```

Comment:
```suggestion
    def try_import(module, abstract_spec=None):
```

Reply:
This is a very minor point, but I would leave it as it is. The point is that this method needs to be called from another and always with two arguments. The API interface for any client of `spack.bootstrap` should be:

https://github.com/spack/spack/blob/2e647d40f55affdc7eaf69cb93cbc869c724fca7/lib/spack/spack/bootstrap.py#L292

which instead has the signature you suggest above.

Reply:
In that case, you should remove the line `abstract_spec = abstract_spec or module` from the first line of all three `try_import` methods.

Reply:
Good catch! Done in f8b13ea